Atomic/pull.py

```diff
@@ -18,20 +19,27 @@ def cli(subparser):
                        help=_("Specify the storage. Default is currently '%s'.  You can"
                               " change the default by editing /etc/atomic.conf and changing"
                               " the 'default_storage' field." % backend))
+    pullp.add_argument("-t", "--type", dest="reg_type", default=None,
+                       help=_("Push to an alternative registry type."))
     pullp.add_argument("image", help=_("image id"))
 
 
 class Pull(Atomic):
     def pull_docker_image(self):
-        _, _, tag = decompose(self.args.image)
+        registry, _, tag = decompose(self.args.image)
+        insecure = True if is_insecure_registry(self.d.info()['RegistryConfig'], remove_port(registry)) else False
         # If no tag is given, we assume "latest"
         tag = tag if tag != "" else "latest"
-        fq_name = skopeo_inspect("docker://{}".format(self.args.image))['Name']
+        if self.args.reg_type == "atomic":
+            pull_uri = 'atomic:'
```

Comment:
The original motivation was simply that the `//` is too much mandatory=meaningless noise and extra typing.

Strictly speaking URI format is schema:schema-specific something; `//` is used only when the part starts with an “authority” (server name/address), and our original design didn’t have any hostnames in `atomic:`.  Now that `docker:` and `atomic:` use the same docker API endpoint + repo etc. semantics, being consistent about `//` would make more sense…

… but then I’ve been tempted for a long time to make the `//` in `docker://` optional :)